### PR TITLE
Allow sdk packages

### PR DESCRIPTION
**Pull request by jonasfj**
Fixing the issue where packages with SDK-dependencies can't be published.

CC @jakemac53 (I believe we'll have a workaround for `_macros` out today'ish; this will ship thursday or next week)

--------

Noticed some slow tests and ended up filing:
https://github.com/dart-lang/pub-dev/issues/7608

### CODE REVIEW

File: app/test/package/upload_test.dart [hunk 1]
```diff
@@ -1049,6 +1049,32 @@ void main() {
...
+          code: 'PackageRejected',
+          message: 'Dependency `abc` does not exist.',
+        );
+      });
+
+      testWithProfile('has SDK-dependency', fn: () async {
+        final tarball = await packageArchiveBytes(
+            pubspecContent: generatePubspecYaml('xyz', '1.2.3') +
+                '  my_sdk_dep:\n'
+                    '    sdk: dart\n');
```
Comment (isoos) [inline app/test/package/upload_test.dart:1070]: This will allow package to have any dependency as long as they have `sdk: dart` (or `sdk:flutter`)?

**Comment by isoos** [comment]
> Should we remove the special-case allow-listing of the macros package also?

We should probably have a tool or test that checks allow-listing regularly and indicates which to remove.

**Comment by jonasfj** [comment]
As far as SDK packages are concerned this will allow dependency on any SDK package.

I think that's fine, you can pick one that doesn't exist, and then the package won't work for anyone. That's sad, but probably people will test their package first :see_no_evil: 

We could make a list of allowed SDK-dependencies. But I'm not sure it's necessary.

**Comment by jakemac53** [comment]
LGTM, will the other fix (to add `_macros` to the reserved name list) not fix the problem for publishing then? Or do you think it will?

**Comment by jakemac53** [comment]
> That fix will allow you to publish `_macros`

Did you mean `macros`? Fwiw, I still can't publish right now, but I don't know if it has been deployed yet.

**Comment by isoos** [comment]
> Did you mean macros? Fwiw, I still can't publish right now, but I don't know if it has been deployed yet.

We were not able to deploy it yet. (needed to roll back)

**Pull request merged**